### PR TITLE
[pornoxo] Improve metadata extraction

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### CODE REVIEW

File: youtube_dl/extractor/pornoxo.py [hunk 1]
```diff
@@ -26,22 +28,17 @@ class PornoXOIE(InfoExtractor):
...
-
-        description = self._html_search_regex(
-            r'<meta name="description" content="([^"]+)\s*featuring',
-            webpage, 'description', fatal=False)
-
-        thumbnail = self._html_search_regex(
-            r'\'image\'\s*:\s*"([^"]+)"', webpage, 'thumbnail', fatal=False)
+        video_url = video_data['file']
+        title = self._html_search_regex(r'<h1>([^<]+)</h1>',
+            webpage, 'title')
```
Comment (yan12125) [inline youtube_dl/extractor/pornoxo.py]: This statement does not conform to PEP8.

Comment (TRox1972) [inline youtube_dl/extractor/pornoxo.py]: fixed

File: youtube_dl/extractor/pornoxo.py [hunk 2]
```diff
@@ -26,22 +28,16 @@ class PornoXOIE(InfoExtractor):
...
-        video_id = mobj.group('id')
+        video_id, display_id = mobj.groups()
 
         webpage = self._download_webpage(url, video_id)
 
-        video_url = self._html_search_regex(
-            r'\'file\'\s*:\s*"([^"]+)"', webpage, 'video_url')
+        video_data = self._parse_json(self._search_regex(
+            r'(?s)jwplayer\("mediaplayer"\)\.setup\(({[^;]+})\);',
+            webpage, 'video data'), video_id, transform_source=js_to_json)
```
Comment (yan12125) [inline youtube_dl/extractor/pornoxo.py]: Looks like another JWPlayer site. Use ```_extract_jwplayer_data``` instead.

File: youtube_dl/extractor/pornoxo.py [hunk 3]
```diff
@@ -26,22 +28,16 @@ class PornoXOIE(InfoExtractor):
...
-            r'<title>([^<]+)\s*-\s*PornoXO', webpage, 'title')
-
-        description = self._html_search_regex(
-            r'<meta name="description" content="([^"]+)\s*featuring',
-            webpage, 'description', fatal=False)
-
-        thumbnail = self._html_search_regex(
-            r'\'image\'\s*:\s*"([^"]+)"', webpage, 'thumbnail', fatal=False)
+        video_url = video_data['file']
+        title = self._html_search_regex(r'<h1>([^<]+)</h1>', webpage, 'title')
```
Comment (yan12125) [inline youtube_dl/extractor/pornoxo.py]: Is the original regex broken?

Comment (TRox1972) [inline youtube_dl/extractor/pornoxo.py]: No, but `<title>` contains more than the title. Should I rely on the old method, and have this as a fallback?

File: youtube_dl/extractor/pornoxo.py (hunk 2 above)
Comment (TRox1972) [inline youtube_dl/extractor/pornoxo.py]: using `_find_jwplayer_data`

File: youtube_dl/extractor/pornoxo.py [hunk 4]
```diff
@@ -26,22 +28,16 @@ class PornoXOIE(InfoExtractor):
...
-        video_id = mobj.group('id')
+        video_id, display_id = mobj.groups()
 
         webpage = self._download_webpage(url, video_id)
 
-        video_url = self._html_search_regex(
-            r'\'file\'\s*:\s*"([^"]+)"', webpage, 'video_url')
+        video_data = self._parse_json(
+            self._find_jwplayer_data(webpage), video_id,
+            transform_source=js_to_json)
```
Comment (yan12125) [inline youtube_dl/extractor/pornoxo.py]: ```_parse_jwplayer_data``` should also be used. See ```tvnoe.py``` for an example.

File: youtube_dl/extractor/pornoxo.py (hunk 3 above)
Comment (yan12125) [inline youtube_dl/extractor/pornoxo.py]: ```<h1>``` is more likely to change than ```<title>```, so I guess extracting from ```<title>``` is better.

Comment (TRox1972) [inline youtube_dl/extractor/pornoxo.py]: reversed

File: youtube_dl/extractor/pornoxo.py (hunk 4 above)
Comment (TRox1972) [inline youtube_dl/extractor/pornoxo.py]: done

**Comment by yan12125** [comment]
Thanks. Could you update the commit message? Now not only metadata extraction is changed.

**Comment by TRox1972** [comment]
@yan12125 done

**Comment by yan12125** [comment]
Good job!

**Pull request merged**